docs/intro/tutorial07.txt

```diff
@@ -0,0 +1,422 @@
+
+    from .models import Question
+
+
+    class QuestionAdmin(admin.ModelAdmin):
+        fieldsets = [
+            (None,               {'fields': ['question_text']}),
+            ('Date information', {'fields': ['pub_date']}),
+        ]
+
+    admin.site.register(Question, QuestionAdmin)
+
+The first element of each tuple in
+:attr:`~django.contrib.admin.ModelAdmin.fieldsets` is the title of the fieldset.
+Here's what our form looks like now:
+
+.. image:: _images/admin08t.png
+   :alt: Form has fieldsets now
+
+Adding related objects
+======================
+
+OK, we have our Question admin page. But a ``Question`` has multiple ``Choices``, and
+the admin page doesn't display choices.
+
+Yet.
+
+There are two ways to solve this problem. The first is to register ``Choice``
+with the admin just as we did with ``Question``. That's easy:
+
+.. snippet::
+    :filename: polls/admin.py
+
+    from django.contrib import admin
+
+    from .models import Choice, Question
+    # ...
+    admin.site.register(Choice)
+
+Now "Choices" is an available option in the Django admin. The "Add choice" form
+looks like this:
+
+.. image:: _images/admin09.png
+   :alt: Choice admin page
+
+In that form, the "Question" field is a select box containing every question in the
+database. Django knows that a :class:`~django.db.models.ForeignKey` should be
+represented in the admin as a ``<select>`` box. In our case, only one question
+exists at this point.
+
+Also note the "Add Another" link next to "Question." Every object with a
+``ForeignKey`` relationship to another gets this for free. When you click "Add
+Another", you'll get a popup window with the "Add question" form. If you add a question
+in that window and click "Save", Django will save the question to the database and
+dynamically add it as the selected choice on the "Add choice" form you're
+looking at.
+
+But, really, this is an inefficient way of adding ``Choice`` objects to the system.
+It'd be better if you could add a bunch of Choices directly when you create the
+``Question`` object. Let's make that happen.
+
+Remove the ``register()`` call for the ``Choice`` model. Then, edit the ``Question``
+registration code to read:
+
+.. snippet::
+    :filename: polls/admin.py
+
+    from django.contrib import admin
+
+    from .models import Choice, Question
+
+
+    class ChoiceInline(admin.StackedInline):
+        model = Choice
+        extra = 3
+
+
+    class QuestionAdmin(admin.ModelAdmin):
+        fieldsets = [
+            (None,               {'fields': ['question_text']}),
+            ('Date information', {'fields': ['pub_date'], 'classes': ['collapse']}),
+        ]
+        inlines = [ChoiceInline]
+
+    admin.site.register(Question, QuestionAdmin)
+
+This tells Django: "``Choice`` objects are edited on the ``Question`` admin page. By
+default, provide enough fields for 3 choices."
+
+Load the "Add question" page to see how that looks:
+
+.. image:: _images/admin10t.png
```

Comment:
this screenshot needs to be redone due to the removal of "collapse" from "Date information".